test/baseWallet.js

```diff
@@ -13,55 +16,94 @@ describe("Test BaseWallet", function () {
+            await wallet.init(owner.address, [module1.contractAddress]);
+            walletOwner = await wallet.owner();
+            assert.equal(walletOwner, owner.address, "owner should be the owner after init");
+        });
+
+        it("should create a wallet with the correct modules", async () => {
+            await wallet.init(owner.address, [module1.contractAddress, module2.contractAddress]);
+            let module1IsAuthorised = await wallet.authorised(module1.contractAddress);
+            let module2IsAuthorised = await wallet.authorised(module2.contractAddress);
+            let module3IsAuthorised = await wallet.authorised(module3.contractAddress);
+            assert.equal(module1IsAuthorised, true, "module1 should be authorised");
+            assert.equal(module2IsAuthorised, true, "module2 should be authorised");
+            assert.equal(module3IsAuthorised, false, "module3 should not be authorised");
+        });
+
+        it("should accept ETH", async () => {
+            let before = await deployer.provider.getBalance(wallet.contractAddress);
+            await nonowner.sendTransaction({ to: wallet.contractAddress, value: 50000000 });
+            let after = await deployer.provider.getBalance(wallet.contractAddress);
+            assert.equal(after.sub(before).toNumber(), 50000000, "should have received ETH");
+        });
 
-    it("should accept ETH", async () => {
-        let before = await deployer.provider.getBalance(wallet.contractAddress);
-        await nonowner.sendTransaction({ to: wallet.contractAddress, value: 50000000 });
-        let after = await deployer.provider.getBalance(wallet.contractAddress);
-        assert.equal(after.sub(before).toNumber(), 50000000, "should have received ETH");
+        it("should delegate static calls to the modules", async () => {
+            await wallet.init(owner.address, [module1.contractAddress]);
+            let module1IsAuthorised = await wallet.authorised(module1.contractAddress);
+            assert.equal(module1IsAuthorised, true, "module1 should be authorised");
+            let walletAsModule = deployer.wrapDeployedContract(Module, wallet.contractAddress);
+            let boolVal = await walletAsModule.contract.getBoolean();
+            let uintVal = await walletAsModule.contract.getUint();
+            let addressVal = await walletAsModule.contract.getAddress(nonowner.address);
+            assert.equal(boolVal, true, "should have the correct bool");
+            assert.equal(uintVal, 42, "should have the correct uint");
+            assert.equal(addressVal, nonowner.address, "should have the address");
+        });
+
+    })
+
+    describe("New BaseWallet", () => {
+        it("should work with old modules", async () => {
+            await wallet.init(owner.address, [oldModule.contractAddress]);
+            await oldModule.callDapp(wallet.contractAddress);
```

Comment:
As mentioned above I think we should test the new wallet interacting with a dapp that returns a value using an old module.

Reply:
Done